### PR TITLE
DML documentation

### LINKED ISSUE #7319: Document DML statements
### Is your feature request related to a problem or challenge?

DataFusion now supports basic `INSERT` and `COPY` statements. 

### Describe the solution you'd like

We should update https://arrow.apache.org/datafusion/user-guide/sql/index.html to reflect this

Examples are in https://github.com/apache/arrow-datafusion/blob/main/datafusion/sqllogictest/test_files/copy.slt and https://github.com/apache/arrow-datafusion/blob/main/datafusion/sqllogictest/test_files/insert_to_external.slt

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by spaydar**
## Which issue does this PR close?

Closes #7319.

## Are these changes tested?

Yes, I can successfully build and view documentation locally as per ["Build & Preview" instructions](https://github.com/apache/arrow-datafusion/blob/main/docs/README.md#build--preview)

### CODE REVIEW

File: docs/source/user-guide/sql/dml.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+where <i><b>option</i></b> can be one of:
+    FORMAT <i><b>format_name</i></b>
+    PER_THREAD_OUTPUT <i><b>boolean</i></b>
+    ROW_GROUP_SIZE <i><b>integer</i></b>
+    ROW_GROUP_LIMIT_BYTES <i><b>integer</i></b>
+</pre>
+
+```sql
+> COPY source_table TO 'table' (FORMAT parquet, PER_THREAD_OUTPUT true);
```
Comment (Dandandan) [inline docs/source/user-guide/sql/dml.md:37]:
```suggestion
> COPY source_table TO 'filename.parquet' (FORMAT PARQUET, PER_THREAD_OUTPUT TRUE);
```

File: docs/source/user-guide/sql/dml.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+## COPY
+
+Copy a table to file(s). Supported file formats are `parquet`, `csv`, and `json`.
+
+<pre>
+COPY <i><b>table_name</i></b> TO '<i><b>file_name</i></b>' [ ( <i><b>option</i></b> [, ... ] ) ]
+
+where <i><b>option</i></b> can be one of:
+    FORMAT <i><b>format_name</i></b>
```
Comment (Dandandan) [inline docs/source/user-guide/sql/dml.md:30]: Can we add the supported formats as well?

File: docs/source/user-guide/sql/dml.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+## INSERT
+
+Insert values into a table.
+
+<pre>
+INSERT INTO <i><b>table_name</i></b> { VALUES ( <i><b>expression</i></b> [, ...] ) [, ...] | <i><b>query</i></b> }
+</pre>
+
+```sql
+> INSERT INTO source_table VALUES (1, 'Foo'), (2, 'Bar');
```
Comment (Dandandan) [inline docs/source/user-guide/sql/dml.md:54]:
```suggestion
> INSERT INTO target_table VALUES (1, 'Foo'), (2, 'Bar');
```

File: docs/source/user-guide/sql/dml.md (hunk 2 above)
Comment (spaydar) [reply docs/source/user-guide/sql/dml.md:32]: Supported formats are already specified on line 24.

File: docs/source/user-guide/sql/dml.md (hunk 1 above)
Comment (spaydar) [reply docs/source/user-guide/sql/dml.md]: I added [an explanation and another example](https://github.com/apache/arrow-datafusion/pull/7362/commits/487ffbf4d6e9e39990e1d187ac7605441a743fc7) to clarify when one should specify a file name or a directory name.

**Review by alamb — APPROVED** [review]
Thanks @spaydar  -- this is a great initial step! I am sure we'll add more content to these pages, but not at least there is a place to do so.

cc @devinjdangelo

**Comment by alamb** [comment]
I noticed we didn't have docs for `COPY (SELECT ...)` so I propose extending these docs in https://github.com/apache/arrow-datafusion/pull/7388

**Pull request merged**